pkg/allow/allow_labels.go

```diff
@@ -49,11 +51,23 @@ func (a Labels) Allowed(metric string, labels, values []string) ([]string, []str
 
 	var finalLabels, finalValues []string
 	labelSet := labelSet(allowedLabels)
+	alreadyMatched := make(map[string]interface{}) // used to prevent duplicate labels
 	for _, allowedLabel := range labelSet {
 		for i, label := range labels {
-			if label == allowedLabel {
+			_, ok := alreadyMatched[label]
+			if ok {
+				continue
+			}
+			allowedLabelReg, err := regexp.Compile(allowedLabel)
+			if err != nil {
+				klog.Error(err)
```

Comment:
I think we should check if the regex are valid right away and exit if the configuration is invalid. This will avoid users running a misconfigured ksm without noticing. IIRC the `allow-deny-list` flag already has a similar mechanism.

Reply:
Perfect, i'll take a look!